QuadradiusR/scripts/game_manager.gd

```diff
@@ -13,16 +13,28 @@ func _ready():
 # # # # # # # # #
 func _game_state_diff(data: Dictionary):
 	var diff = data['game_state_diff']
-	if diff['current_player_id'] == Context.user_id:
-		# Play opponent's move
-		if 'pieces' in diff['board']:
-			for piece_id in diff['board']['pieces'].keys():
-				if piece_id[0] == "$":
-					continue
-				var dest_tile_id = diff['board']['pieces'][piece_id]['tile_id']
-				var source_tile_pos = Context.game_state.get_tile_pos_by_piece_id(piece_id)
-				var dest_tile_pos = Context.game_state.get_tile_pos_by_tile_id(dest_tile_id)
-				board.move_torus_by_tiles(source_tile_pos, dest_tile_pos)
+	if 'pieces' in diff['board']:
+		for piece_id in diff['board']['pieces'].keys():
+			if piece_id[0] == "$":
```

Comment:
What does `$` mean?

Reply:
The server marks data that was deleted in this way :D this means we need to remove it from gamestate